lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -264,8 +300,9 @@ void ThreadPoolExecutor::executeDAGNode(
 
   if (deviceManagerIt == deviceManagers_.end()) {
     // Mark the node as no longer executing.
-    executionState->decrementInflightNodes();
     executionState->setResultCode(ResultCode::Failed);
+    executionState->decrementInflightNodes();
```

Comment:
IIRC you made these executionState inflight counters atomic to avoid having to lock while changing them, but I think with the lock around the class level inflight counter it's actually less efficient for these to be atomic rather than locked since it will create multiple memory barriers.